numba/smartarray.py

```diff
@@ -92,33 +96,41 @@ def ndim(self): return self._ndim
     @property
     def size(self): return self._size
 
-    def host(self):
-        """Return the host representation of 'self'."""
-
-        self._sync('host')
-        return self._host
-
-    def host_changed(self):
-        """Mark the host array as changed, broadcast updates if needed."""
-
-        self._invalidate('gpu')
-        # only sync if there are active views
-        if self._gpu is not None and sys.getrefcount(self._gpu) > 2:
-            self._sync('gpu')
+    def get(self, where='host'):
+        """Return the representation of 'self' in the given memory space."""
 
-    def gpu(self):
-        """Return the GPU representation of 'self'."""
+        if where not in self._targets:
+            raise ValueError('"%s" is not a valid target'%where)
+        self._sync(where)
+        if where == 'host': return self._host
+        elif where == 'gpu': return self._gpu
+        else: raise ValueError('unknown memory space "%s"'%where)
 
-        self._sync('gpu')
-        return self._gpu
+    @deprecated("get('host')")
+    def host(self): return self.get('host')
+    @deprecated("get('gpu')")
+    def gpu(self): return self.get('gpu')
 
-    def gpu_changed(self):
-        """Mark the gpu array as changed, broadcast updates if needed."""
+    def mark_changed(self, where='host'):
+        """Mark the given location as changed, broadcast updates if needed."""
 
-        self._invalidate('host')
-        # only sync if there are active views
-        if self._host is not None and sys.getrefcount(self._host) > 2:
-            self._sync('host')
+        if where not in self._targets:
+            raise ValueError('"%s" is not a valid target'%where)
+        if where == 'host':
+            self._invalidate('gpu')
+            # only sync if there are active views
+            if self._gpu is not None and sys.getrefcount(self._gpu) > 2:
```

Comment:
If anyone refers to the array it needs to be synced, I don't see any way around that.